### DISCUSSION TITLE
make -j error: unknown type name ‘OSSL_ENCRYPTION_LEVEL’

### DISCUSSION THREAD

**Discussion by hbf731eF** [firedancer-io/firedancer#888] [Support]
following doc/getting-started.md on a Vanilla CentOS Stream 9
`$ uname -snr`
Linux centos9 5.14.0-375.el9.x86_64

`$ dnf groupinstall development`
`$ git clone https://github.com/firedancer-io/firedancer.git`
`$ cd firedancer`
`$ make -j`
[...]
src/tango/quic/tls/fd_quic_tls.c:41:37: error: unknown type name ‘OSSL_ENCRYPTION_LEVEL’
[...]

`$ grep OSSL_ENCRYPTION_LEVEL /usr/include/openssl/ssl.h`

`$ rpm -qf /usr/include/openssl/ssl.h`
openssl-devel-3.0.7-24.el9.x86_64

**Comment by ripatel-fd** [comment]
@hbf731eF It looks like you are missing the dependencies step. Please try the following:

```
dnf groupinstall development
git clone https://github.com/firedancer-io/firedancer.git
cd firedancer
./deps.sh
make -j
```

Please note that Firedancer does not support upstream OpenSSL.
We currently use the [quictls fork](https://github.com/quictls/openssl), which adds support for the QUIC-TLS encryption layer.

**Comment by ripatel-fd** [comment]
Also, thanks for volunteering to test on CentOS Stream 9. We appreciate it 🙏

**Reply by hbf731eF** [reply]
thanks for quick response! 
`$ dnf install perl-CPAN`
`$  ./deps.sh`
`$ make -j`

 gcc: fatal error: Killed signal terminated program cc1
compilation terminated.
make: *** [config/everything.mk:273: build/native/gcc/obj/ballet/reedsol/wrapped_impl/fd_reedsol_ppt_impl_55.o] Error 1
gcc: fatal error: Killed signal terminated program cc1
compilation terminated.

are there memory requirements when calling make?

`cat /proc/meminfo`
MemTotal:        3899020 kB

**Reply by hbf731eF** [reply]
you need 10GB of RAM to run make -j

**Reply by mmcgee-jump** [reply]
Yup, that's definitely an OOM kill. fd_reedsol_ppt_impl is compiling some pretty heavy generated code. Might have lower memory requirements with a non-parallel make (without the `-j`)

**Comment by mmcgee-jump** [comment]
https://github.com/firedancer-io/firedancer/issues/894

**Comment by ripatel-fd** [comment]
@hbf731eF Were you able to compile it? Please let me know if you ran into any other issues.

**Comment by hbf731eF** [comment]
commit a2106f7d22a774b is the last referencing frank 

README :
     * Moved the frank tile pieces into fdctl and removed all references to
       "frank" in the codebase. Frankendancer remains only in the README, to
       describe the current side-by-side model on main.

the examples are therefore not working, because after building scripts are missing 

`$ grep frank src/wiredancer/README.md`
+ Configure app-frank
  - `sudo ./build/linux/gcc/x86_64/bin/fd_frank_init_demo frank 1-6 ./build/linux/gcc/x86_64 /tmp/solana.pcap 0 0 1 0`
+ Run app-frank
  - `sudo ./build/linux/gcc/x86_64/bin/fd_frank_run frank "1-6"`
  - `sudo taskset -c 7 build/linux/gcc/x86_64/bin/wd_frank_mon frank --duration 0`
  - `sudo taskset -c 7 build/linux/gcc/x86_64/bin/fd_frank_mon frank --duration 10e12 --dt-min 1e7 --dt-max 1e7`    
      
      
  please name the proper  `fd_*_ctl`   replacement commands in `src/wiredancer/README.md`

**Comment by ripatel-fd** [comment]
Thanks, let's open a separate issue regarding frank.

You can find an older (working) demo image of the FPGA on `milestone-1.4-demo`. The wiredancer code on `main` is currently broken in a Frankendancer setup.

**Comment by mmcgee-jump** [comment]
Filed #911 for the wiredancer changes needed for main.
